File: 1_topicos_iniciais/csharp/Leran/Leran/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Leran
{
    internal class Product
    {
        public string Name { get; set; }
        public double Price { get; set; }
        public int Quantity { get; set; }

        public Product(string name, double price, int quantity) { 
            Name = name;
            Price = price;
            Quantity = quantity;
        }

        public override string ToString()
        {
            return Name +" | " + " R$" + Price + " | " + Quantity;
        }

        public double total()
        {
            return Price * Quantity;
        }
        public void updatePrice(double percentage)
        {
            Price = Price * (1.0 + percentage / 100.0);
        }
    }
}

File: 1_topicos_iniciais/java/Learn/src/main/java/org/acounts/Product.java
package org.acounts;

public class Product {
    private   String name;
    private  double price;
    private int quantity;

    public Product(String name, double price, int quantity){
        this.name = name;
        this.price = price;
        this.quantity = quantity;
    }

    public double total(){
        return price * quantity;
    }

    public void updatePrice (double percentage){
        price = price * (1.0 + percentage / 100.0);
    }

    @Override
    public String toString() {
        return name + ": " + String.format("%.2f", price) + ": " + quantity;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }
}
